File: client-clojure/src/sirqul_io_t_platform/specs/wrapped_proxy_response.clj
(ns sirqul-io-t-platform.specs.wrapped-proxy-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def wrapped-proxy-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :returning) string?
   (ds/opt :type) string?
   (ds/opt :responseCode) int?
   (ds/opt :responseRaw) string?
   (ds/opt :requestDuration) int?
   (ds/opt :requestEndTime) int?
   })

(def wrapped-proxy-response-spec
  (ds/spec
    {:name ::wrapped-proxy-response
     :spec wrapped-proxy-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/WrappedProxyResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import WrappedProxyResponse._

case class WrappedProxyResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
returning: Option[String],
`type`: Option[String],
responseCode: Option[Integer],
responseRaw: Option[String],
requestDuration: Option[Long],
requestEndTime: Option[Long])

object WrappedProxyResponse {
  import DateTimeCodecs._

  implicit val WrappedProxyResponseCodecJson: CodecJson[WrappedProxyResponse] = CodecJson.derive[WrappedProxyResponse]
  implicit val WrappedProxyResponseDecoder: EntityDecoder[WrappedProxyResponse] = jsonOf[WrappedProxyResponse]
  implicit val WrappedProxyResponseEncoder: EntityEncoder[WrappedProxyResponse] = jsonEncoderOf[WrappedProxyResponse]
}
